examples/kotlin-smartapp/src/main/kotlin/routes/SmartApp.kt

```diff
@@ -70,6 +75,11 @@ fun Route.smartAppExecution() {
          *     Kotlin can automatically infer the types required,
          *     making this code much more succinct.
          */
-        call.respond(smartApp.execute(call.receive()))
+        val executionRequest: ExecutionRequest = call.receive()
+        if (executionRequest.lifecycle == AppLifecycle.PING || httpVerificationService.verify(call.request)) {
+            call.respond(smartApp.execute(executionRequest))
+        } else {
+            call.respond(HttpStatusCode.Unauthorized, "unable to verify request")
+        }
```

Comment:
Tangential thought - I wish that we didn't have to force developers to know _"handle PING without verification, but also then handle verification before handling an execution request.. otherwise the request isn't authorized and respond with a 401"_ but I do recognize that our SDKs would need to be deeply integrated with the HTTP request. On top of that, this entire signature verification process may become deprecated. 🤷‍♂️

Reply:
Yeah, that went through my mind too when I was setting up the first example. There probably is something better we could do (perhaps with a lambda). Hmm...